### ISSUE TITLE
[TF2] The Iron bomber Reload Animation Quirk.

### ISSUE THREAD

**Issue by VirtualSoda** [ValveSoftware/Source-1-Games#2135]
**Upon reloading The Iron Bomber; The Demoman misses the Chamber position, and loads of of his Grenades into the same Chamber (Supposedly)**

![iron-bomber](https://cloud.githubusercontent.com/assets/25712612/22943811/1fe1a00c-f2ef-11e6-9fd4-60c3cc7fc93f.gif)

**Comment by DRAG4YOU** [comment]
Here's something that you guys can do to fix this:

The Iron bomber and the Loose Cannon have been using the Grenade Launcher Reload Animation for quite some time causing clipping with the bomb not going inside the chamber when reloading. A solution for this, will be to have an iron bomber/loose cannon unique reload animation only just like the Loch-n-Load has.

This would benefit the game because there's a mod that fixes this issue for these grenade launchers [Definitive Grenade Launcher Reload](https://gamebanana.com/mods/494356)

https://youtu.be/deRyClZAUfI
https://youtu.be/NS6dYh0eVnA

BUT if someone is using a 4 slot grenade launcher it causes a little bit of clipping at the insert of the grenade.

https://youtu.be/pRpgQQlrimQ

I made a 4 slot grenade launcher reload animation base on the tf2c one to help with this but because these 3 grenade launchers use the same reload it still causes the grenade clipping issue.

https://youtu.be/nqY-5B3PAig

A way to fix this is to add 3 new sequences in the c_demo_animations.mdl .qc like this:

$sequence "iron_reload_start" {
"c_demo_animations_anims\iron_reload_start.smd"
activity "ACT_PRIMARY_RELOAD_START_3" 1
{ event 5004 5 "Weapon_GrenadeLauncher.DrumOpen" }
{ event AE_CL_BODYGROUP_SET_VALUE_CMODEL_WPN 1 "reload 1" }
fadein 0.2
fadeout 0.2
snap
fps 30
}

$sequence "iron_reload_loop" {
"c_demo_animations_anims\iron_reload_loop.smd"
activity "ACT_PRIMARY_VM_RELOAD_3" 1
{ event 5004 6 "Weapon_GrenadeLauncher.DrumLoad" }
{ event AE_CL_BODYGROUP_SET_VALUE_CMODEL_WPN 1 "reload 1" }
{ event AE_CL_BODYGROUP_SET_VALUE_CMODEL_WPN 20 "reload 0" }
fadein 0.2
fadeout 0.2
snap
fps 30
}

$sequence "iron_reload_end" {
"c_demo_animations_anims\iron_reload_end.smd"
activity "ACT_PRIMARY_RELOAD_FINISH_3" 1
{ event 5004 5 "Weapon_GrenadeLauncher.DrumClose" }
{ event AE_CL_BODYGROUP_SET_VALUE_CMODEL_WPN 1 "reload 0" }
fadein 0.2
fadeout 0.2
snap
fps 30
}

And inside the c_demo_animations_anims folder copy and paste these smds g_reload_start.smd, g_reload_loops.smd, g_reload_end.smd and with the copies rename them to iron_reload_start.smd, iron_reload_loops.smd, iron_reload_end.smd

These 3 new sequences are very similar to the loch-n-load/grenade launcher reload ones but they have a _3, an unused activity that is in the game but it is unused.

In other words have the Grenade Launcher only use the g_reload_start, g_reload_loop, g_reload_end and iron bomber/loose cannon use the new sequences above. The animations of these sequences are the same as the grenade launchers ones to facilitate the creation of this and modification of these for further fixing.

I know these seems unnecessary in the way that we have Suodoi's mod that can fix this but its been quite some time that the Grenade Launcher has lost its 6 grenades to 4 and the model of the grenade launcher hasn't been updated in the game to address this, but having this will help us modders fix this simple issue.
